### ISSUE TITLE
ISequence can return a Stream from the current position

### ISSUE THREAD

**Issue by Whiteknight** [Whiteknight/ParserObjects#133]
Every ISequence which operates on `char` or `byte` should be able to return a `Stream` for reading contents from the current position. This stream should have the rules:
1. The stream should not be rewindable to a negative value. The returned stream should not give any access to input items which have already been consumed.
1. Items read from the Stream should advance the underlying sequence accordingly (either in real-time, or after a `.Dispose()` call, TBD)

With this we should be able to write adaptors for existing non-PO parsers like System.Text.Json, Newtonsoft.Json and System.Data.Xml

**Comment by Whiteknight** [comment]
I have started to implement this, but there are two concerns which are stopping development on this feature for now:

1. Stream.Seek() causes problems with char-based sequences because we lose track of line/column/consumed metadata if we seek to a random position in the stream. We would need to store metadata milestones in the sequence for every N chars, and then calculate the metadata if we Seek to a position between milestones. This is `O(N)` to calculate (where N is probably smallish) and `O(1/N)` memory consumption, which may still be an issue for very large inputs
2. NewtonSoft.Json and System.Text.Json do not seem to support this use-case. System.Text.Json doesn't seem to support UTF-16 streams at all, or non-json suffixes. Newtonsoft likewise seems to consume the entire stream even when we set `SupportMultipleContent=true` and `CheckAdditionalContent=false` in the parser. This means we can't combine with other parsers because there's no input left in the sequence.

I eventually need to try to implement Seek() on my streams, to see if the readers will return unused buffered characters to the stream.
